### PR TITLE
pianoteq.{stage-trial,standard-trial,stage-6,stage-7}: init at 7.3.0

### PR DESCRIPTION
**Pull request by mausch**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/pianoteq/default.nix [hunk 1]
```diff
@@ -0,0 +1,146 @@
+{ pkgs ? (import <nixpkgs> {}) }:
+with pkgs;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pianoteq/default.nix:2]:
```suggestion
{ ... }:
```

please specify the packages explicitly.

Comment (mausch) [reply pkgs/applications/audio/pianoteq/default.nix]: Done in e520318a0050cb0c8ecee9b04009f38082a9b0c6 , thanks

**Comment by mausch** [comment]
Not sure why this was marked as "work in progress"? From my point of view this is ready to be merged... 
If that's about the TODO notes those are just hints to improve things in the future or add more versions / platforms.

**Comment by alesguzik** [comment]
A hash for stage-7 is below:
```
nix-prefetch-url file:///home/me/Downloads/pianoteq_stage_linux_v730.7z
[29.5 MiB DL]
path is '/nix/store/pzz8kc7idq921a05164h85kkjj7crmc2-pianoteq_stage_linux_v730.7z'
05w7sv9v38r6ljz9xai816w5z2qqwx88hcfjm241fvgbs54125hx
```

(BTW, if you own stage-6 the upgrade is just 29 EUR, not full 129)

**Comment by alesguzik** [comment]
@ofborg build pianoteq.stage-trial pianoteq.standard-trial

**Comment by mausch** [comment]
@alesguzik Thanks! I just tried installing the trial again and it looks like they've changed their pages, so the basic scraping I was doing here no longer works. It was bound to happen sooner or later I guess. 
Their new pages have some javascript manipulation and while it doesn't look too hard to figure out (they've not even minified it) it's pretty clear they're doing this to prevent scrapers, so I'd rather not try to work around things.

So unfortunately we'll have to install from a local file. I see [that's what they're doing over at Arch too](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=pianoteq-stage).

**Comment by mausch** [comment]
@alesguzik I fixed the fetchers and some issues with runtime dependencies in 7.x .
I've tested stage-trial, standard-trial and stage-6 . Can you try stage-7 ?

**Comment by alesguzik** [comment]
@mausch stage-7 works well, thanks

**Comment by alesguzik** [comment]
@ofborg build pianoteq.stage-trial pianoteq.standard-trial

**Comment by alesguzik** [comment]
I also checked stage-trial and standard-trial and they are surprisingly useful, as they include all their instruments, not just ones you bought.

**Comment by alesguzik** [comment]
@mausch could you please change the commit text to match what ofborg expects: https://github.com/NixOS/ofborg

File: pkgs/applications/audio/pianoteq/default.nix [hunk 2]
```diff
@@ -0,0 +1,200 @@
+{ lib, stdenv, curl, gnugrep, jq, xorg, alsaLib, freetype, p7zip, autoPatchelfHook, writeShellScript, zlib, libjack2, makeWrapper }:
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pianoteq/default.nix:1]:
```suggestion
{ lib, stdenv, curl, gnugrep, jq, xorg, alsa-lib, freetype, p7zip, autoPatchelfHook, writeShellScript, zlib, libjack2, makeWrapper }:
```

File: pkgs/applications/audio/pianoteq/default.nix [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+      nativeBuildInputs = [
+        autoPatchelfHook
+        makeWrapper
+      ];
+
+      buildInputs = [
+        stdenv.cc.cc.lib
+        xorg.libX11      # libX11.so.6
+        xorg.libXext     # libXext.so.6
+        alsaLib          # libasound.so.2
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/pianoteq/default.nix:24]:
```suggestion
        alsa-lib # libasound.so.2
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/pianoteq/default.nix (hunk 2 above)
Comment (mausch) [reply pkgs/applications/audio/pianoteq/default.nix]: applied in d040aff89639da4383473cb7afdd3e98d579b164

File: pkgs/applications/audio/pianoteq/default.nix (hunk 3 above)
Comment (mausch) [reply pkgs/applications/audio/pianoteq/default.nix]: applied in d040aff89639da4383473cb7afdd3e98d579b164

**Comment by mausch** [comment]
Updated the alias thing and the title of the PR.

**Comment by alesguzik** [comment]
@mausch it should be in the commit message in git, not in github PR name.

**Comment by mausch** [comment]
pffft done in https://github.com/NixOS/nixpkgs/pull/91254/commits/3fbc8ee273086dce176c2c0e58d9d8aeebd0ee37

**Comment by alesguzik** [comment]
@SuperSandro2000 is there anything else that needs to be done? Should we remove the work-in-progress tag and then merge it?

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 is there anything else that needs to be done? Should we remove the work-in-progress tag and then merge it?

I am undecided if nixpkgs is really the place for this and it does not seem to be complete either.

**Comment by mausch** [comment]
Hi @SuperSandro2000 , thanks for your reviews, can you explain why nixpkgs isn't the place for this and what's missing for the package to become complete?

**Comment by seb314** [comment]
> > @SuperSandro2000 is there anything else that needs to be done? Should we remove the work-in-progress tag and then merge it?
> 
> I am undecided if nixpkgs is really the place for this and it does not seem to be complete either.

For me at least its pretty nice to have this simple way for installing this package (thanks @mausch !!), and re completeness the package worked flawlessly out-of-the-box.

**Comment by alesguzik** [comment]
I also find it very useful, and this package also works out of the box for me. We already package various commercial audio software (e.g. Reaper, Renoise, Bitwig Studio), so I believe it belongs here just as much. Pianoteq even works as an LV2 plugin, not just as a VST.

**Comment by alesguzik** [comment]
@mausch actually there's a problem with the package. Right now it wraps the `.so` file, so it doesn't work as a VST plugin, only as an app. Also, VST and LV2 plugins should be installed under `/lib/vst` and `/lib/lv2` , but right now they go into `/bin`.

**Comment by mausch** [comment]
Yes, this isn't perfect. Other things that could be fixed / improved:

- Refactor to expose the functions in here so people can easily build their own packages e.g. from a local file with a different version / hash etc.
- Mac support
- LAME clashes
- LAME exports probably don't work (for the same reason, need to replace it with a nixed LAME)
- Desktop items (i.e. using `makeDesktopItem`)

If people expect this to be perfect it will likely never get merged.
But merging it does not mean development will stop, on the contrary it could mean more people using it and contributing to it to scratch their own itch (I don't have a personal need to improve any of the above tbh)

Up to you folks. I'm not likely to spend more time on this tbh.

**Pull request merged**